### PR TITLE
fix: project pipeline Reduce status queries and create precheck

### PR DESCRIPTION
**Pull request by kakj-go**
#### What this PR does / why we need it:
bugfix project pipeline Reduce status queries
bugfix project pipeline create precheck same name definition

#### Which issue(s) this PR fixes:

- [Erda Cloud Issue Link](https://terminus-org.app.terminus.io/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb24iOls4ODMsNzcyXSwibWVtYmVyIjpbIjEwMDA1NjAiXX0%3D&id=275799&iterationID=772&pId=0&type=BUG)

- [Erda Cloud Issue Link](https://terminus-org.app.terminus.io/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb24iOls4ODMsNzcyXSwibWVtYmVyIjpbIjEwMDA1NjAiXX0%3D&id=276321&iterationID=772&pId=0&type=BUG)

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |       project pipeline Reduce status queries and create precheck       |
| 🇨🇳 中文    |      项目流水线减少状态查询并创建预检查        |

### CODE REVIEW

File: modules/dop/providers/projectpipeline/service_impl.go [hunk 1]
```diff
@@ -171,6 +179,48 @@ func (p *ProjectPipelineService) Create(ctx context.Context, params *pb.CreatePr
...
+	if err != nil {
+		return false, err
+	}
+	if len(appList.List) == 0 {
+		return false, fmt.Errorf("not find app")
+	}
+
+	appDto, err := p.bundle.GetApp(appID)
+	if err != nil {
+		return false, err
```
Comment (Effet) [inline modules/dop/providers/projectpipeline/service_impl.go:197]: optimize it, too slow

**Review by Effet — CHANGES_REQUESTED** [review]

File: modules/dop/component-protocol/components/project-pipeline/myPage/tabsTable/filterContainer/customFilter/provider.go [hunk 2]
```diff
@@ -128,8 +130,20 @@ func (p *CustomFilter) RegisterRenderingOp() (opFunc cptype.OperationFunc) {
...
+		var realSearchStatus []string
+		for _, status := range values.Status {
+			realSearchStatus = append(realSearchStatus, func() []string {
+				var statusString []string
+				for _, v := range util.PipelineDefinitionStatusMapList[apistructs.PipelineStatus(status)] {
+					statusString = append(statusString, v.String())
+				}
+				return statusString
+			}()...)
+		}
```
Comment (Effet) [inline modules/dop/component-protocol/components/project-pipeline/myPage/tabsTable/filterContainer/customFilter/provider.go:143]: separate a func like transferStatus

**Review by Effet — DISMISSED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**